qcore/validate_vm.py

```diff
@@ -15,219 +15,251 @@
-    Equation taken from http://www.edwilliams.org/avform.htm#Int
-    :param lon_lat1: A tuple containing the longitude and latitude of the start point
-    :param lon_lat2: A tuple containing the longitude and latitude of the end point
-    :param lon_in: A float or iterable of floats compliant with numpy of the longitude(s) to have latitudes calculated for
-    :return: A float or iterable of floats which represent the latitude(s) of the value(s) given in lon_in
+    Validates that a velocity model file has the correct size, and no 0 values
+    :param file_name: A Path object representing the file to test
+    :param vm_size: The size of the VM
+    :return: A possibly empty list of issues with the VM file
     """
-    conversion_factor = np.pi / 180
-    lon1, lat1 = lon_lat1
-    lon2, lat2 = lon_lat2
-    lat1 *= conversion_factor
-    lon1 *= conversion_factor
-    lat2 *= conversion_factor
-    lon2 *= conversion_factor
-    lon = lon_in * conversion_factor
-    if lon1 == lon2:
-        return np.linspace(lat1, lat2, len(lon_in))/conversion_factor
-    return (
-        np.arctan(
-            (
-                np.sin(lat1) * np.cos(lat2) * np.sin(lon - lon2)
-                - np.sin(lat2) * np.cos(lat1) * np.sin(lon - lon1)
-            )
-            / (np.cos(lat1) * np.cos(lat2) * np.sin(lon1 - lon2))
+    errors = []
+    size = file_name.stat().st_size
+    if size != vm_size * SIZE_FLOAT:
+        errors.append(
+            f"VM filesize for {file_name} expected: {vm_size * SIZE_FLOAT} found: {size}"
+        )
+    if (
+        not np.min(
+            np.fromfile(file_name, dtype="<f{}".format(SIZE_FLOAT), count=vm_size)
         )
-    ) / conversion_factor
+        > 0
+    ):
+        errors.append(f"File {file_name} has minimum value of 0.0")
+    return errors
 
 
-if __name__ == "__main__":
-    rc = 1
+def main():
     parser = argparse.ArgumentParser()
-    parser.add_argument("VM_dir", type=str, help="path the VM folder")
-    parser.add_argument("--srf", default=None, type=str, help="Path to the SRF to be used with this VM")
+
+    sub_parser = parser.add_subparsers(dest="subparser_name")
+
+    vm_params_parser = sub_parser.add_parser(
+        PARAMS_SUB_PARSER,
+        help="Validate a vm_params.yaml file. If the srf is given, will test that the srf is contained within the VM.",
+    )
+    vm_params_parser.add_argument(
+        "vm_params", type=Path, help="Path to vm_params.yaml file"
+    )
+    vm_params_parser.add_argument(
+        "srf", type=Path, help="Path to srf file", nargs="?", default=None
+    )
+
+    vm_file_parser = sub_parser.add_parser(
+        NZVM_SUB_PARSER, help="Validate files generated by the NZVM"
+    )
+    vm_file_parser.add_argument("vm_dir", type=Path, help="path the VM folder")
+
+    pert_parser = sub_parser.add_parser(
+        SINGLE_FILE_SUB_PARSER,
+        help="Validate a single VM file for size and 0s. Primarily intended for perturbation files.",
```

Comment:
"file" sounds a little misleading, especially in presence of "files". While the option names are singular vs plural, what they actually do are very different. If this is only intended for sanity check including size and non-zero values, what about "sanity"? Could this be handling multiple files too?

Reply:
I have been thinking about better names. I was thinking of 'NVZM' instead of 'files', as it's all based on the NZVM output. 
For our current use case I think leaving 'file' with its current name is ok, and only doing one file.
The entire script is a sanity check for VM files, just validating them and making sure they seem reasonable, so wouldn't be useful as an option name

Reply:
nzvm sounds good. What about "quick" for "file"?

Reply:
Quick isn't necessarily a quick test. It's a test to check a single output file from the NZVM. So maybe a NZVM_binary or similar? 

Or we could call them: complete and perturbation as those are the use cases we are using them with. But they can be applied more generally than that.

Reply:
as it only checks the size and if it contains non-zero values, "quick"er than full-scale check

Reply:
It's not there to be a quick check, it's there to check a single file such as a perturbation or Qp/Qs file that isn't in the regular NZVM output, so I don't think the name 'quick' is appropriate. I'm not that keen on calling it NZVM_binary when the main use case is non NZVM binaries
My current preferred names are 'params', 'NZVM', 'file'

Reply:
I don't think the naming is a big issue as the scripts call will often be automated. I think most of the suggestions here have made sense; so happy with your judgement.

Reply:
I can settle on "params, NZVM, file".